File: 13_function_overriding.py
class Shape:
    def area(self):
        pass

class Square(Shape):
    def __init__(self, side):
        self.side = side

    def area(self):
        return self.side * self.side

class Circle(Shape):
    def __init__(self, radius):
        self.radius = radius

    def area(self):
        return 3.1416 * self.radius * self.radius


square = Square(4)
circle = Circle(3)

print(square.area())
print(circle.area())

File: 13_function_overriding.cpp
#include<bits/stdc++.h>

using namespace std;
#define fast ios_base::sync_with_stdio(0); cin.tie(NULL);
#define ll long long

class Shape{
public:
    void area(){}
};

class Square:public Shape{
    int side;
public:
    Square(int side)
    {
        this->side = side;
    }

    void area()
    {
        cout<<side*side<<endl;
    }
};

class Circle:public Shape{
    int radius;
public:
    Circle(int radius)
    {
        this->radius = radius;
    }

    void area()
    {
        cout<<3.1416*radius*radius<<endl;
    }
};

void solve()
{
    Square square(3);
    Circle circle(3);

    square.area();
    circle.area();
}

int main()
{
    fast;
    solve();
    return 0;
}